app/gui/src/controller/searcher.rs

```diff
@@ -628,7 +439,8 @@ impl Searcher {
         let def_id = graph.graph().id;
         let def_span = double_representation::module::definition_span(&module_ast, &def_id)?;
         let module_repr: Rope = module_ast.repr().into();
-        let position = module_repr.offset_to_location_snapped(def_span.end);
+        // TODO: is it correct?
+        let position = module_repr.offset_to_location_snapped(def_span.end - text::ByteDiff(1));
```

Comment:
Subtracting one byte from position helps to find local variables declared inside `main` function. Apparently the engine thinks that locals are out of scope otherwise and doesn't send them in `completion` response. This issue should be properly investigated separately.

Reply:
Let's make a GH issue for that and put its number here.

Reply:
I fixed the issue completely by searching position before creating a new node.